handlers/handler-auth/src/main/java/io/servicecomb/authentication/consumer/RSACoumserTokenManager.java

```diff
@@ -0,0 +1,77 @@
+package io.servicecomb.authentication.consumer;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Date;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import io.servicecomb.authentication.RSAAuthenticationToken;
+import io.servicecomb.foundation.common.utils.RSAUtils;
+import io.servicecomb.foundation.token.AuthenticationTokenManager;
+import io.servicecomb.foundation.token.RSAKeypair4Auth;
+import io.servicecomb.serviceregistry.RegistryUtils;
+
+public class RSACoumserTokenManager implements AuthenticationTokenManager {
+
+	private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+	
+	private RSAAuthenticationToken token;
+
+	@Override
+	public String getToken() {
+		readWriteLock.readLock().lock();
+		if(null != token && valid(token.fromat()))
+		{
+			String tokenStr = token.fromat();
+			readWriteLock.readLock().unlock();
+			return tokenStr;
+		}
+		else
+		{
+			readWriteLock.readLock().unlock();
+			return createToken();
```

Comment:
if expired, maybe create multiple token in different thread in the same time.
if this is allowed, then the lock control is useless.

Reply:
accept